plugins/modules/vmware_guest_powerstate.py

```diff
@@ -0,0 +1,559 @@
+            
+    def configure_vm_answerable_powerstate(self):
+        """
+        Configures a VM powerstate when question_answers option is set
+        """
+        try:
+            responses = self.make_answer_response(self.module.params['question_answers'])
+            self.answer_question(responses)
+        except Exception as e:
+            self.module.fail_json(msg="%s" % e)
+
+        # Wait until a virtual machine is unlocked
+        while True:
+            if not (hasattr(self.vm, "runtime") and self.vm.runtime.question):
+                break
+
+        self.result['changed'] = True
+    
+    def current_state_matches_desired_state(self):
+        """
+        Checks the hosts current power state and compares it to the desired power state setting.
+        Returns:
+            bool, true if they match, otherwise false
+        """
+        if self.current_state == self.desired_state:
+            return True
+        
+        return False
+
+
+def main():
+    module = AnsibleModule(
+        argument_spec={
+            **base_argument_spec(), **dict(
+                datacenter=dict(type='str', required=True),
+                state=dict(type='str', default='powered-on',
+                        choices=['powered-off', 'powered-on', 'reboot-guest', 'restarted', 'shutdown-guest', 'suspended']),
+                name=dict(type='str'),
+                name_match=dict(type='str', choices=['first', 'last'], default='first'),
+                uuid=dict(type='str'),
+                moid=dict(type='str'),
+                use_instance_uuid=dict(type='bool', default=False),
+                folder=dict(type='str', required=False),
+                force=dict(type='bool', default=False),
+                scheduled_at=dict(type='str', required=False),
+                scheduled_task_name=dict(type='str', required=False),
+                scheduled_task_description=dict(type='str', required=False),
+                scheduled_task_enabled=dict(type='bool', default=True),
+                timeout=dict(type='int', default=0),
+                question_answers=dict(type='list',
+                            required=False,
+                            elements='dict',
+                            options=dict(
+                                question=dict(type='str', required=True),
+                                response=dict(type='str', required=True)
+                            ))
+            )
+        },
+        supports_check_mode=True,
+        mutually_exclusive=[
+            ['name', 'uuid', 'moid'],
+            ['scheduled_at', 'question_answers']
+        ],
+        required_one_of=[
+            ['name', 'uuid', 'moid']
+        ],
+    )
+
+    result = dict(
+        changed=False,
+        result={}
+    )
+
+    vmware_guest_powerstate = VmwareGuestPowerstateModule(module)
+    vmware_guest_powerstate.standardize_folder_param()
+    if vmware_guest_powerstate.current_state_matches_desired_state():
```

Comment:
```suggestion
    if vmware_guest_powerstate.current_state == vmware_guest_powerstate.desired_state:
```
this method is redundant